src/mlpack/core/data/string_encoding.hpp

```diff
@@ -0,0 +1,197 @@
+/**
+ * @file string_encoding.hpp
+ * @author Jeffin Sam
+ *
+ * Definition of the StringEncoding class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_STRING_ENCODING_HPP
+#define MLPACK_CORE_DATA_STRING_ENCODING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/boost_backport/boost_backport_string_view.hpp>
+#include <mlpack/core/data/string_encoding_dictionary.hpp>
+#include <mlpack/core/data/string_encoding_policies/policy_traits.hpp>
+#include <vector>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * Definition of the StringEncoding class. The class translates a set of
+ * strings into numbers using various encoding algorithms.
+ *
+ * @tparam EncodingPolicyType Type of the encoding algorithm itself.
+ * @tparam DictionaryType Type of the dictionary.
+ */
+template<typename EncodingPolicyType,
+         typename DictionaryType>
+class StringEncoding
+{
+ public:
+  /**
+   * Pass the given arguments to the policy constructor and cunstruct
```

Comment:
I think this should be "construct" instead of "cunstruct".

Reply:
It's already resolved in #1960.